cursor.go

```diff
@@ -261,13 +261,27 @@ func (c *Cursor) Fetch(db SqlPreparer) (batch *RowBatch, paginationKeypos uint64
 		values:             batchData,
 		paginationKeyIndex: paginationKeyIndex,
 		table:              c.Table,
+		columns:            filterByTableColumns(columns, c.Table.Columns),
 	}
 
 	logger.Debugf("found %d rows", batch.Size())
 
 	return
 }
 
+func filterByTableColumns(columnsToFilter []string, tableColumns []schema.TableColumn) (out []string) {
+	exists := make(map[string]struct{}, len(tableColumns))
+	for _, c := range tableColumns {
+		exists[c.Name] = struct{}{}
+	}
+	for _, c := range columnsToFilter {
+		if _, ok := exists[c]; ok {
+			out = append(out, c)
+		}
```

Comment:
Why do we need this filter? Also this seems like the columns coming from the query could be a subset of tables.Columns? Shouldn't it match exactly otherwise we should raise an error?

Reply:
It's simply to filter out the MD5 column if fingerprinting is on.